lib/ruby_indexer/lib/ruby_indexer/index_rbs.rb

```diff
@@ -0,0 +1,45 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RubyIndexer
+  class IndexRbs
+    extend T::Sig
+
+    sig { params(index: Index).void }
+    def initialize(index)
+      @index = index
+    end
+
+    sig { void }
+    def index_core_classes
+      loader = RBS::EnvironmentLoader.new
+      RBS::Environment.from_loader(loader).resolve_type_names
+
+      loader.each_signature do |source, pathname, _buffer, declarations, _directives|
+        process_signature(source, pathname, declarations)
+      end
+    end
+
+    private
+
+    sig { params(source: T.untyped, pathname: Pathname, declarations: T::Array[RBS::AST::Declarations::Base]).void }
+    def process_signature(source, pathname, declarations)
+      declarations.each do |declaration|
+        process_declaration(declaration, pathname)
+      end
+    end
+
+    sig { params(declaration: RBS::AST::Declarations::Base, pathname: Pathname).void }
+    def process_declaration(declaration, pathname)
+      return unless declaration.is_a?(RBS::AST::Declarations::Class)
+
+      nesting = [declaration.name.name.to_s]
+      file_path = pathname.to_s
+      location = RubyIndexer::Location.from_rbs_location(declaration.location)
+      comments = Array(declaration.comment&.string&.lines)
+      parent_class = nil
```

Comment:
Is this `nil` variable needed? Can we not get the parent class?

Reply:
You should be able to access it from the declaration: https://github.com/ruby/rbs/blob/64a96b134a9c2e4441d389f3895e3dcdba9f8938/lib/rbs/ast/declarations.rb#L92

Reply:
Thanks, updated.